File: bind/cpp/hello.cpp
// hello.cpp
//
// PDFlib client: hello example in C++
//
//

#include <stdio.h>
#include <stdlib.h>

#include "pdflib.hpp"

int
main(void)
{
    PDF *p;			// pointer to the PDF class
    int font;

    p = new PDF();

    // Open new PDF file
    if (p->open("hello_cpp.pdf") == -1) {
	fprintf(stderr, "Error: cannot open PDF file hello_cpp.pdf.\n");
	exit(2);
    }

    p->set_info("Creator", "hello.cpp");
    p->set_info("Author", "Thomas Merz");
    p->set_info("Title", "Hello, world (C++)!");

    // start a new page
    p->begin_page((float) a4_width, (float) a4_height);

    font = p->findfont("Helvetica-Bold", "default", 0);
    if (font == -1) {
	fprintf(stderr, "Couldn't set font!\n");
	exit(3);
    }

    p->setfont(font, 24);

    p->set_text_pos(50, 700);
    p->show("Hello, world!");
    p->continue_text("(says C++)");
    p->end_page();				// finish page

    p->close();					// close PDF document
    delete p;

    exit(0);
}

File: bind/java/hello.java
/* hello.java
 * Copyright (C) 1997-99 Thomas Merz. All rights reserved.
 *
 * PDFlib client: hello example in Java
 */

public class hello
{
    public static void main (String argv[])
    {
	long p;
	int font;

	p = pdflib.PDF_new();

	if (pdflib.PDF_open_file(p, "hello_java.pdf") == -1) {
	    System.err.println("Couldn't open PDF file hello_java.pdf\n");
	    System.exit(1);
	}

	pdflib.PDF_set_info(p, "Creator", "hello.java");
	pdflib.PDF_set_info(p, "Author", "Thomas Merz");
	pdflib.PDF_set_info(p, "Title", "Hello world (Java)");

	pdflib.PDF_begin_page(p, 595, 842);

	font = pdflib.PDF_findfont(p, "Helvetica-Bold", "default", 0);
	if (font == -1){
	    System.err.println("Couldn't find font!\n");
	    System.exit(1);
	}

	pdflib.PDF_setfont(p, font, 18);

	pdflib.PDF_set_text_pos(p, 50, 700);
	pdflib.PDF_show(p, "Hello world!");
	pdflib.PDF_continue_text(p, "(says Java)");
	pdflib.PDF_end_page(p);

	pdflib.PDF_close(p);
	pdflib.PDF_delete(p);
    }
}
